### PR TITLE
SIDAM compliant functional tests for SAAT

### PR DESCRIPTION
**Pull request by linusnorton**
Update the functional tests to use a SIDAM compliant method of getting an auth token.

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/reform/dg/template/management/testutil/Env.java [hunk 1]
```diff
@@ -27,22 +21,6 @@ public static String getTestUrl() {
         return require("TEST_URL");
     }
 
-    public static String getIdamURL() {
```
Comment (linusnorton) [inline src/aat/java/uk/gov/hmcts/reform/dg/template/management/testutil/Env.java:30]: These are no longer necessary as they are loaded via the application.yaml

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -2,15 +2,6 @@ spring:
   application:
     name: DG TEMPLATE MANAGEMENT API
 
-base-urls:
```
Comment (linusnorton) [inline src/main/resources/application.yaml:5]: These were never used

File: build.gradle [hunk 3]
```diff
@@ -68,13 +68,17 @@ dependencies {
     compile group: 'commons-validator', name: 'commons-validator', version: '1.6'
     compile group: 'commons-io', name: 'commons-io', version: '2.6'
     compile group: 'uk.gov.hmcts.reform.auth', name: 'auth-checker-lib', version: '2.1.3'
-    compile group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '2.0.0'
+    compile group: 'uk.gov.hmcts.reform', name: 'idam-client', version: '0.0.5'
```
Comment (linusnorton) [inline build.gradle]: Very useful find :D

File: build.gradle [hunk 4]
```diff
@@ -68,13 +68,17 @@ dependencies {
...
     compile group: 'uk.gov.hmcts.reform', name: 'java-logging-appinsights', version: '5.0.1'
+    compile group: 'uk.gov.hmcts.reform', name: 'java-logging-spring', version: '3.0.4'
+    compile group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '2.0.0'
     compile "org.mapstruct:mapstruct-jdk8:1.3.0.Final"
     annotationProcessor "org.mapstruct:mapstruct-processor:1.3.0.Final"
     annotationProcessor "org.springframework.boot:spring-boot-configuration-processor"
 
-    testCompile "org.springframework.boot:spring-boot-starter-test"
+    testCompile('org.springframework.boot:spring-boot-starter-test') {
+        exclude group: "com.vaadin.external.google", module:"android-json"
```
Comment (linusnorton) [inline build.gradle:79]: fix a json lib warning

File: build.gradle [hunk 5]
```diff
@@ -83,7 +87,7 @@ dependencies {
 bootJar {
     baseName 'rpa-dg-template-management-api'
     archiveName = "$baseName"+".jar"
-    destinationDir = file("$rootDir/build/libs")
+    destinationDir = file("$rootDir/build/lib")
```
Comment (linusnorton) [inline build.gradle:89]: When running the gradle assembly job a shell script is created (I don't know where it comes from) it references the jar in `lib` not `libs` so I updated it to match

File: src/main/resources/application.yaml [hunk 6]
```diff
@@ -65,11 +56,15 @@ app-insights:
...
-  base-url: ${DG_TEMPLATE_MANAGEMENT_API_URL:http://localhost:4623}
-
+# only used in functional tests
 idam:
   s2s-auth:
     url: ${S2S_BASE_URI:http://localhost:4502}
-    totp_secret: ${S2S_KEY:AAAAAAAAAAAAAAAA}
-    microservice: dg_template_management
+    totp_secret: ${FUNCTIONAL_TEST_CLIENT_S2S_TOKEN:AAAAAAAAAAAAAAAA}
+    microservice: em_gw
```
Comment (linusnorton) [inline src/main/resources/application.yaml]: Looking at the changes I'm not 100% sure whether this is used at runtime. If it is I will need to separate it from the functional test config

File: build.gradle (hunk 3 above)
Comment (JeroenRijks) [reply build.gradle]: :+1:

File: build.gradle (hunk 4 above)
Comment (JeroenRijks) [reply build.gradle:79]: TimJ previously said we don't need java-logging-spring. Do we know for sure that we need it now?

Comment (linusnorton) [reply build.gradle:79]: seems to be required, will chase it up later

**Pull request closed without merging**